cuda_core/cuda/core/experimental/_utils.py

```diff
@@ -134,3 +141,17 @@ def get_device_from_ctx(ctx_handle) -> int:
         assert ctx_handle == handle_return(cuda.cuCtxPopCurrent())
         handle_return(cuda.cuCtxPushCurrent(prev_ctx))
     return device_id
+
+
+def is_list_or_tuple(obj):
+    """
+    Check if the given object is a sequence (list or tuple).
+    """
+    return isinstance(obj, (list, tuple))
+
+
+def is_nested_list_or_tuple(obj):
+    """
+    Check if the given object is a nested sequence (list or tuple with atleast one list or tuple element).
+    """
+    return is_list_or_tuple(obj) and any(is_list_or_tuple(elem) for elem in obj)
```

Comment:
ditto, also, it should be `all`:
```suggestion
    return is_list_or_tuple(obj) and all(is_list_or_tuple(elem) for elem in obj)
```

Reply:
I think perhaps I should take this out of utils and leave it in program if it is too specific, but to handle the case in question it should be any imo, since I would like to support define_macro = ("MY_MACRO", ("MY_OTHER_MACRO", 99))

Reply:
btw since we've changed to check `Sequence`, this should be `all` (because of our unwanted side effect that `str` is considered a `Sequence`).